package.json

```diff
@@ -12,28 +12,35 @@
         "debug": ">=0.8.0"
     },
     "devDependencies": {
+        "@types/node": "^8.0.37",
+        "@types/debug": "^0.0.30",
+        "@types/mocha": "^5.2.0",
         "browserify": "",
         "eslint": "",
-        "istanbul": "",
+        "istanbul": "1.1.0-alpha.1",
         "minimist": "",
         "mocha": "^3.0.0",
         "uglify-js": "",
         "require-self": "^0.2.1",
-        "ws": "^5.1.1"
+        "ws": "^5.1.1",
+        "typescript": "^2.8.3",
+        "ts-node": "^6.0.3"
     },
     "scripts": {
         "lint": "eslint lib/*.js",
-        "test": "mocha",
-        "coverage": "istanbul cover _mocha",
+        "test": "mocha -r ts-node/register ./test/*.ts",
+        "coverage": "istanbul cover _mocha -- test/*.ts --compilers ts-node/register",
```

Comment:
bumping the version of istanbul and making this change in coverage got the test coverage working. [This answer on SO](https://stackoverflow.com/a/37908693/2967423) helped in getting it working. The ts-node compiler transpiles the .ts test files to .js in memory and runs those tests. Hence we need not run tests twice in ts and js.